deploy/manifests/scc.yaml

```diff
@@ -0,0 +1,37 @@
+# Create SCC for Gateway resources
+kind: SecurityContextConstraints
+apiVersion: security.openshift.io/v1
+metadata:
+  name: nginx-ngf-admin
+allowPrivilegedContainer: false
+allowHostDirVolumePlugin: false
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPorts: false
+readOnlyRootFilesystem: true
+runAsUser:
+  type: MustRunAsRange
+  uidRangeMin: 101
+  uidRangeMax: 102
+fsGroup:
+  type: MustRunAs
+  ranges:
+    - min: 1001
+      max: 1001
+supplementalGroups:
+  type: MustRunAs
+  ranges:
+    - min: 1001
+      max: 1001
+seLinuxContext:
+  type: MustRunAs
+volumes:
+  - emptyDir
+  - secret
+users:
+  - 'system:serviceaccount:*:ngf-nginx-gateway-fabric'
```

Comment:
I think we need to add a helm template for this file since the service account name may differ across deployments.

Reply:
I like the idea of using Helm to template the file - it means we can remove the manual step of installing the SCC separately. We can also add a manifest generation step using helm templating for manifest installation.

The reason NIC does it a bit differently (i.e. not using Helm to generate the file) is because the Operator is used, which creates NIC installations which all share the SCC.

If we are adding a Helm template for this file, we may need to make generating this object as part of the install optional while still supporting OpenShift - so maybe accepting the name of the SCC as a parameter for cases where the object already exists on an OpenShift cluster and should be used instead of creating a new one and using that.

Reply:
Good to know. What is the best way to express that in the helm chart?

Reply:
Added some the helm template with some dynamic fields. Here is the output of 
`helm template ngf ./charts/nginx-gateway-fabric --create-namespace -n nginx-gateway --set nginxGateway.productTelemetry.enable=false --set platform=openshift --set sccName=my-name --set nginxGateway.securityContext.allowPrivilegeEscalation=true --set fullnameOverride=helloooo`

```
# Create SCC for Gateway resources
kind: SecurityContextConstraints
apiVersion: security.openshift.io/v1
metadata:
  name: my-name
allowPrivilegeEscalation: true
readOnlyRootFilesystem: true
runAsUser:
  type: MustRunAsRange
  uidRangeMin: 101
  uidRangeMax: 102
fsGroup:
  type: MustRunAs
  ranges:
    - min: 1001
      max: 1001
supplementalGroups:
  type: MustRunAs
  ranges:
    - min: 1001
      max: 1001
seLinuxContext:
  type: MustRunAs
volumes:
  - emptyDir
  - secret
users:
  - system:serviceaccount:*:helloooo
allowedCapabilities:
  - NET_BIND_SERVICE
  - KILL
requiredDropCapabilities:
  - ALL
```

When I don't include the platform like such: `helm template ngf ./charts/nginx-gateway-fabric --create-namespace -n nginx-gateway --set nginxGateway.productTelemetry.enable=false` 

The SCC yaml file is not generated.

And here is the output of `helm template ngf ./charts/nginx-gateway-fabric --create-namespace -n nginx-gateway --set nginxGateway.productTelemetry.enable=false --set platform=openshift` 

```
# Source: nginx-gateway-fabric/templates/scc.yaml
# Create SCC for Gateway resources
kind: SecurityContextConstraints
apiVersion: security.openshift.io/v1
metadata:
  name: nginx-ngf-admin
allowPrivilegeEscalation: false
readOnlyRootFilesystem: true
runAsUser:
  type: MustRunAsRange
  uidRangeMin: 101
  uidRangeMax: 102
fsGroup:
  type: MustRunAs
  ranges:
    - min: 1001
      max: 1001
supplementalGroups:
  type: MustRunAs
  ranges:
    - min: 1001
      max: 1001
seLinuxContext:
  type: MustRunAs
volumes:
  - emptyDir
  - secret
users:
  - system:serviceaccount:*:ngf-nginx-gateway-fabric
allowedCapabilities:
  - NET_BIND_SERVICE
  - KILL
requiredDropCapabilities:
  - ALL
```
which looks correct.

Reply:
> Good to know. What is the best way to express that in the helm chart?

Maybe we could add a `createSCC` flag that defaults to `true`, and then in our scc template, have 

`{{- if and (eq .Values.platform "openshift") .Values.createSCC }}`

(or maybe something more explicit that this only for OpenShift like `createOpenShiftSCC`)

The sccName field has already been added, so the correct RBAC rules will still be applied when the platform is OpenShift even when the SCC is being created externally.

Reply:
@sjberman with the recent changes of creating the name through the helpers template, is there a way to still support the SCC being created externally (I know we can add a createSCC flag, but the name is sort of tucked away in the corner and would be tough for them to configure without changing the names of all the crds)? 

Do we even want to support that, I'm not sure of any use cases?

Reply:
If we include it in our manifests, then I think that should be fine, right?

Reply:
Though it would be contained in the single manifest file, not as its own file. I feel like we just do that until we get a request for it to be its own file.

Reply:
Hmm, okay I added the change. Currently if the user wanted to use their own scc file, I think they would add `--set createOpenShiftSCC=false` and they would need to look at the generated rbac for whatever name the clusterrole is binding to and either change that to their SCC or rename their SCC to that name. 

I think.

Reply:
I'm not totally convinced we need to support the case where the resource already exists. I feel like everything should just be contained in our chart, and our docs make it clear that we provide the SCC to attach to NGF. Then we just have the `platform` field. @ciarams87 Do you think this is a common thing where someone provides their own?

Reply:
If we're doing a name unique to the installation, I think it's fine to not support BYO. It just means we'll have multiple SCC resources doing the same thing if there are multiple installations. Which because the `users` field will be unique, is actually better.

Reply:
Will close conversation with recent addition of checking Helm's built-in objects of APICapabilities.